### PR TITLE
feat(packcrack): integrate packcrack into BOINC

### PR DESCRIPTION
**Pull request by BoySanic**
Huzzah! The time has finally come to make an attempt at Pack.PNG.
Fingers crossed that this can have a relatively speedy review.

### CODE REVIEW

**Comment by BoySanic** [comment]
Force-pushed because I never nuked the recaptcha commit from my history

**Review by Earthcomputer — CHANGES_REQUESTED** [review]
Where do the binary files come from?
```
joe@joe-Ubuntu:~/programming/cuda/minecrafthome$ sha256sum ~/Downloads/packcrack.exe 
5d49ab25a6f66a68606439d38ec2d5849981aa78b7b8b59811d14ffda59cb9ba  /home/joe/Downloads/packcrack.exe
joe@joe-Ubuntu:~/programming/cuda/minecrafthome$ sha256sum images/makeproject/project/apps/packcrack/1.01/windows_x86_64__cuda/packcrack_1.01_cuda.exe 
97ec0f8a583a5975979d38dfba91a2aeed2ed3e064e0543c3ff5c3426d274289  images/makeproject/project/apps/packcrack/1.01/windows_x86_64__cuda/packcrack_1.01_cuda.exe
```

**Comment by BoySanic** [comment]
> Where do the binary files come from?
> 
> ```
> joe@joe-Ubuntu:~/programming/cuda/minecrafthome$ sha256sum ~/Downloads/packcrack.exe 
> 5d49ab25a6f66a68606439d38ec2d5849981aa78b7b8b59811d14ffda59cb9ba  /home/joe/Downloads/packcrack.exe
> joe@joe-Ubuntu:~/programming/cuda/minecrafthome$ sha256sum images/makeproject/project/apps/packcrack/1.01/windows_x86_64__cuda/packcrack_1.01_cuda.exe 
> 97ec0f8a583a5975979d38dfba91a2aeed2ed3e064e0543c3ff5c3426d274289  images/makeproject/project/apps/packcrack/1.01/windows_x86_64__cuda/packcrack_1.01_cuda.exe
> ```

https://github.com/Vath0s/pack-sand-reversal/actions/runs/232217178

**Comment by Earthcomputer** [comment]
That's where I downloaded it from, the sha256 of the extracted file doesn't match.

**Comment by BoySanic** [comment]
That's peculiar, because I've just matched it here after downloading both binaries (from Vath0s' repo and from my mc@h repo)

**Comment by MCRcortex** [comment]
I just did a shar and they matched

**Comment by BoySanic** [comment]
PS C:\Users\boysanic> Get-FileHash .\Documents\Github\minecrafthome\images\makeproject\project\apps\packcrack\1.01\windows_x86_64__cuda\packcrack_1.01_cuda.exe                                                                                 
Algorithm       Hash                                                                   Path
---------       ----                                                                   ----
SHA256          5D49AB25A6F66A68606439D38EC2D5849981AA78B7B8B59811D14FFDA59CB9BA       C:\Users\boysanic\Documents\G...


PS C:\Users\boysanic> Get-FileHash .\Downloads\packcrack\packcrack.exe                                                  
Algorithm       Hash                                                                   Path
---------       ----                                                                   ----
SHA256          5D49AB25A6F66A68606439D38EC2D5849981AA78B7B8B59811D14FFDA59CB9BA       C:\Users\boysanic\Downloads\p...


PS C:\Users\boysanic> Get-FileHash .\Documents\Github\minecrafthome\images\makeproject\project\apps\packcrack\1.01\x86_64-pc-linux-gnu__cuda\packcrack_1.01_cuda                                                                                
Algorithm       Hash                                                                   Path
---------       ----                                                                   ----
SHA256          7C0C9F37690786F5426F1A58CE754EA22221987172C3571530FDF936CEFD1989       C:\Users\boysanic\Documents\G...


PS C:\Users\boysanic> Get-FileHash .\Downloads\packcrack\packcrack

Algorithm       Hash                                                                   Path
---------       ----                                                                   ----
SHA256          7C0C9F37690786F5426F1A58CE754EA22221987172C3571530FDF936CEFD1989       C:\Users\boysanic\Downloads\p...

**Comment by Earthcomputer** [comment]
I downloaded the windows binary from this PR via the github website, and the sha256 now matches. So git is modifying the file. Do you think it could be replacing newlines?

**Review by MCRcortex — APPROVED** [review]
looks good

**Review by Earthcomputer — CHANGES_REQUESTED** [review]
My issue is now not the checksum, it's that git is corrupting the files. We can't have that. We need to change the gitconfig to treat these files as binary.

**Comment by BoySanic** [comment]
> I downloaded the windows binary from this PR via the github website, and the sha256 now matches. So git is modifying the file. Do you think it could be replacing newlines?

That's interesting. I'm not sure what it's doing, but that is a pretty good guess. I'd assume it wouldn't try to replace things from a binary file, but what do I know. 

What I can say is that on Windows when I git clone the repo, SHA matches. Perhaps Linux is replacing some line endings in the windows binary. It was compiled on Github's windows-latest container.

**Review by Earthcomputer — APPROVED** [review]
It was git LFS not auto-resolving for me, sorry >.<

File: images/makeproject/makeproject-step2.sh [hunk 1]
```diff
@@ -90,6 +90,18 @@ yes | bin/update_versions
 #   --command_line "$((i * 4096)) $(((i + 1) * 4096))" \
 #   --wu_name "${wu_name}"
 #done
+# 2^33 seeds per boinc workunit * 2^15 boinc work units = 2^48 seeds total (PackCrack)
+#for i in {0..32767}; do
+# wu_name="packcrack_1.00_17179869184_$i"
```
Comment (Vath0s) [inline images/makeproject/makeproject-step2.sh:95]: Can we change the name so its consistent with the new work unit size (8589934592)?

**Review by Vath0s — APPROVED** [review]

**Review by Earthcomputer — APPROVED** [review]

**Review by vcokltfre — APPROVED** [review]

**Review by MCRcortex — APPROVED** [review]

**Pull request merged**